nixos/modules/misc/version.nix

```diff
@@ -52,7 +52,7 @@ in
 
     stateVersion = mkOption {
       type = types.str;
-      default = cfg.release;
+      default = warn "system.stateVersion is not set, defaulting to ${cfg.release}. Read why this matters on https://nixos.org/manual/nixos/stable/options.html#opt-system.stateVersion." cfg.release;
```

Comment:
You have to use `config.warnings` instead. Otherwise, building the manual will always produce a warning while evaluating the default.
But I don't see how that's possible right now because the type would have to be changed to something like `nullOr str` which would break all modules using `system.stateVersion`.
Maybe we can use `mkIf` to override the value and produce a warning if it's set to `null`? That might lead to an infinite recursion though.

Reply:
I could create a warning when `options.system.stateVersion.highestPrio` is equal to `1500`, what about that?

Reply:
Sounds good to me! But please document the reasoning in a comment.

Reply:
There's a nice snippet in `nixos/modules/services/continuous-integration/hercules-ci-agent/common.nix`:
```
if options.services.hercules-ci-agent.package.highestPrio == (lib.modules.mkOptionDefault { }).priority
```